include/api.php

```diff
@@ -2974,23 +3025,36 @@ function api_friendica_remoteauth(&$a) {
 	}
 	api_register_func('api/friendica/remoteauth', 'api_friendica_remoteauth', true);
 
-
+	/**
+	 * @brief Return the item shared, if the item contains only the [share] tag
+	 *
+	 * @param array $item Sharer item
+	 * @return array Shared item or false if not a reshare
+	 */
 	function api_share_as_retweet(&$item) {
 		$body = trim($item["body"]);
 
 		// Skip if it isn't a pure repeated messages
 		// Does it start with a share?
 		if (strpos($body, "[share") > 0)
-			return(false);
+			return false;
 
 		// Does it end with a share?
 		if (strlen($body) > (strrpos($body, "[/share]") + 8))
-			return(false);
+			return false;
 
 		$attributes = preg_replace("/\[share(.*?)\]\s?(.*?)\s?\[\/share\]\s?/ism","$1",$body);
-		// Skip if there is no shared message in there
-		if ($body == $attributes)
-			return(false);
+ 		// Skip if there is no shared message in there
+ 		if ($body == $attributes)
+			return false;
+
+		// NOTE: we could check te guid="" attribute and then  try to load original post
```

Comment:
There is this function: https://github.com/friendica/friendica/blob/develop/include/diaspora.php#L2739

it does a check for a repeated item and should be ideal for this. I really should have replaced this function a long time ago ...

See how it is implement here: https://github.com/friendica/friendica/blob/develop/include/diaspora.php#L2027-L2029

Only problem: We maybe should move this function at another place.